File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        print(nums)
        temp=[]
        n=len(nums)
        ans=-1
        temp.append(nums[0])
        for i in range(1,n):
            temp.append(temp[-1]+nums[i])
        for i in range(2,n):
            if(temp[i-1]>nums[i]):
                ans=temp[i]  

        return ans

        

File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.cpp
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        long long ans=-1;
        vector<long long>dm(nums.begin(),nums.end());
        int n=nums.size();
        for(int i=1;i<n;i++)
            dm[i]+=dm[i-1];
        for(int i=2;i<n;i++)
        {
            if(dm[i-1]>nums[i])
                ans=dm[i];
        }
        return ans;
    }
};